media-proxy/src/mesh/st2110.cc

```diff
@@ -0,0 +1,206 @@
+        st_ptime = ST31_PTIME_0_14MS;
+        break;
+    case MESH_AUDIO_PACKET_TIME_0_09MS:
+        st_ptime = ST31_PTIME_0_09MS;
+        break;
+    default:
+        return -1; // Error: unknown packet time
+    }
+
+    return 0; // Success
+}
+
+void *ST2110::get_frame_data_ptr(st_frame *src) {
+    return src->addr[0];
+}
+
+void *ST2110::get_frame_data_ptr(st30_frame *src) {
+    return src->addr;
+}
+
+/**
+ * Use this function in on_establish() to initialize frame_available flag.
+ */
+void ST2110::init_frame_available() {
+    frame_available = false;
+}
+
+/**
+ * Use this function in on_shutdown() and frame_available_cb() to properly notify another thread
+ *
+ */
+void ST2110::notify_frame_available() {
+    frame_available.store(true, std::memory_order_release);
+    frame_available.notify_one();
+}
+
+/**
+ * Use this function in on_receive() to wait for notification
+ *
+ */
+void ST2110::wait_frame_available() {
+    frame_available.wait(false, std::memory_order_acquire);
+    frame_available = false;
+}
+
+int ST2110::frame_available_cb(void *ptr) {
+    auto _this = static_cast<ST2110 *>(ptr);
+    if (!_this) {
+        return -1;
+    }
+
+    _this->notify_frame_available();
+
+    return 0;
+}
+
+void ST2110::get_mtl_dev_params(mtl_init_params& st_param, const std::string& dev_port,
+                                mtl_log_level log_level,
+                                const char local_ip_addr[MESH_IP_ADDRESS_SIZE]) {
+    if (getenv("KAHAWAI_CFG_PATH") == NULL) {
+        setenv("KAHAWAI_CFG_PATH", "/usr/local/etc/imtl.json", 0);
+    }
+    strlcpy(st_param.port[MTL_PORT_P], dev_port.c_str(), MTL_PORT_MAX_LEN);
+    inet_pton(AF_INET, local_ip_addr, st_param.sip_addr[MTL_PORT_P]);
+    st_param.pmd[MTL_PORT_P] = mtl_pmd_by_port_name(st_param.port[MTL_PORT_P]);
+    st_param.num_ports = 1;
+    st_param.flags = MTL_FLAG_BIND_NUMA;
+    st_param.flags |= MTL_FLAG_TX_VIDEO_MIGRATE;
+    st_param.flags |= MTL_FLAG_RX_VIDEO_MIGRATE;
+    st_param.flags |= MTL_FLAG_RX_UDP_PORT_ONLY;
+    st_param.pacing = ST21_TX_PACING_WAY_AUTO;
+    st_param.log_level = log_level;
+    st_param.priv = NULL;
+    st_param.ptp_get_time_fn = NULL;
+    // Native af_xdp have only 62 queues available
+    if (st_param.pmd[MTL_PORT_P] == MTL_PMD_NATIVE_AF_XDP) {
+        st_param.rx_queues_cnt[MTL_PORT_P] = 62;
+        st_param.tx_queues_cnt[MTL_PORT_P] = 62;
+    } else {
+        st_param.rx_queues_cnt[MTL_PORT_P] = 128;
+        st_param.tx_queues_cnt[MTL_PORT_P] = 128;
+    }
+    st_param.lcores = NULL;
+    st_param.memzone_max = 9000;
+}
+
+mtl_handle ST2110::get_mtl_device(const std::string& dev_port, mtl_log_level log_level,
+                                  const char local_ip_addr[MESH_IP_ADDRESS_SIZE], int& session_id) {
+    static mtl_handle dev_handle;
+    static int _session_id;
+    static std::mutex mtx;
+    std::lock_guard<std::mutex> lock(mtx);
+
+    session_id = _session_id++;
```

Comment:
Having this value incremented is not a robust approach to generate an id. I don't like it.

Check if you can use `generate_uuid_v4()` and change the session id type to std::string. See usage example https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/c2f586c7825931470f2074f98bb2a0baed4d2c1e/media-proxy/src/mesh/manager_local.cc#L31

Incremental ids are fine for this PR with the initial implementation, but must be reworked (if possible) in the nearest future to make that production grade.